src/executables/server.c

```diff
@@ -305,6 +308,69 @@ abort_handler (int signo, siginfo_t * siginfo, void *dummyp)
 #endif /* !NDEBUG */
 #endif /* WINDOWS */
 
+static int
+argument_handler (int argc, char **argv)
+{
+  constexpr char *SERVER_TYPE_LONG = "type";
+  constexpr char SERVER_TYPE_SHORT = 't';
+
+  GETOPT_LONG server_options_map[] = {
+    {SERVER_TYPE_LONG, 1, 0, SERVER_TYPE_SHORT},
+    {nullptr, 0, 0, 0}
+  };
+
+  constexpr int SHORT_OPTIONS_BUFSIZE = 64;
+  char short_options_buffer[SHORT_OPTIONS_BUFSIZE];
+
+  utility_make_getopt_optstring (server_options_map, short_options_buffer);
+  while (true)
+    {
+      int option_index = 0;
+      int option_key = getopt_long (argc, argv, short_options_buffer, server_options_map, &option_index);
+      if (option_key == -1)
+	{
+	  break;
+	}
+      switch (option_key)
+	{
+	case SERVER_TYPE_SHORT:
+	  // *INDENT-ON*
+	  if (std::strcmp (optarg, "transaction") == 0)
+	    {
+	      set_server_type (SERVER_TYPE_TRANSACTION);
+	    }
+	  else if (std::strcmp (optarg, "page") == 0)
+	    {
+	      set_server_type (SERVER_TYPE_PAGE);
+	    }
+          // *INDENT-OFF*
+          else
+            {
+              er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, ER_INVALID_SERVER_TYPE_ARGUMENT, 0);	// error that the type is not valid
+              return ER_INVALID_SERVER_TYPE_ARGUMENT;
+            }
+          break;
+        default:
+          er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, ER_INVALID_SERVER_OPTION, 1, optarg);	// invalid server option
+          return ER_INVALID_SERVER_OPTION;
+        }
+    }
+  if (argc - optind == 1)
+    {
+      database_name = argv[optind];
+    }
+  else if (argc > optind)
+    {
+      er_log_debug(ARG_FILE_LINE, "Argument '%s' is not needed for cub_server", argv[optind + 1]); // exit
+    }
+  else
+    {
+      er_log_debug(ARG_FILE_LINE, "A database-name is missing."); // exit
+    }
```

Comment:
A generic invalid argument message may be used here:
```suggestion
  else
    {
      util_log_write_errid (MSGCAT_UTIL_GENERIC_INVALID_ARGUMENT);
      return ER_FAILED;
    }
```